### PR TITLE
implemented homing at endstops

### PR DESCRIPTION
**Pull request by s-bl**
## Description

[//]: # "Description of what you did.  If it is more complex, consider to add a"
[//]: # "'Summary' section."

Added homing at end-stop as an additional homing procedure. 
In this mode, the end-stop position will be used to define zero.  

## How I Tested

Tested with twojoint.yml

## I fulfilled the following requirements

- [x] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

File: config/twojoint.yml
Comment (luator) [inline config/twojoint.yml:7]: It might be a bit clearer what is actually happening if instead of a `homing_at_endstop` option, it would be `homing_with_index` (basically the same but negated).
The current name could be a bit confusing because in both cases it uses the end_stop if `has_endstop == true` and on the other hand it can also be a valid use-case to use it with `had_endstop == false` (meaning "simply home at the current position without moving automatically").

File: include/robot_fingers/n_joint_blmc_robot_driver.hxx
Comment (luator) [inline include/robot_fingers/n_joint_blmc_robot_driver.hxx:118]: This will make it a mandatory option.  For backward-compatibility with existing config files, it would be better to make it optional by wrapping it with a `if (user_config["homing_at_endstop"])`-check.

**Review by luator — CHANGES_REQUESTED** [review]
Thanks a lot for implementing this!
I have a few comments (see above) but it already looks pretty good.

File: config/twojoint.yml
Comment (s-bl) [reply config/twojoint.yml]: Yes, you are right. Much cleaner.

**Review by luator — APPROVED** [review]
Looks great now, thanks a lot!  This will be quite useful for us as well.

I have some comments on the corresponding PR on blmc_drivers, I'll merge here once they are resolved.

**Comment by luator** [comment]
@s-bl I made a small change before merging:  With the flipped meaning of the setting, the default of `homing_with_index` has to be true for compatibility with existing configurations.

**Pull request merged**
